packages/discovery-provider/src/tasks/entity_manager/entities/user.py

```diff
@@ -456,19 +456,17 @@ def add_associated_wallet(
                 f"Invalid signature for wallet {wallet_address}"
             )
 
-        # Check if wallet already exists
+        # Check if wallet already exists, and remove it from other users
         existing_wallet = None
         for _, wallet in params.existing_records["AssociatedWallet"].items():
-            if (
-                wallet.chain == chain
-                and wallet.user_id == user_id
-                and wallet.wallet == wallet_address
-            ):
-                existing_wallet = wallet
-                break
+            if wallet.chain == chain and wallet.wallet == wallet_address:
+                if wallet.user_id == user_id:
+                    existing_wallet = wallet
+                else:
+                    session.delete(wallet)
```

Comment:
Feels like we could just do a "delete where" on the wallet address, no?

Reply:
probably, but sticking with consistent conventions for now